### PR TITLE
Added Dead letter queue  and upgrade to Kafka Connect 2.8.0

### PR DESCRIPTION
**Pull request by jomach**
This Pr adds a dead letter queue

### CODE REVIEW

**Comment by nervgh** [comment]
DLQ is one of most important features in my opinon. Thanks @jomach 

So I vote for this PR :+1: 

Here below I've copy-pasted some info about this feature from official docs:

> ### [KIP-610: Error Reporting in Sink Connectors](https://cwiki.apache.org/confluence/display/KAFKA/KIP-610%3A+Error+Reporting+in+Sink+Connectors)
> 
> Extends an Error Reporting capability for sink connectors. Previously, error handling in Kafka Connect included functionality such as retrying, logging, and sending errant records to a dead letter queue. However, the dead letter queue functionality from [KIP-298](https://cwiki.apache.org/confluence/display/KAFKA/KIP-298%3A+Error+Handling+in+Connect) only supports error reporting within contexts of the transform operation, and key, value, and header converter operation. After records are sent to the connector for processing, there is no support for dead letter queue or error reporting functionality. [KIP-610](https://cwiki.apache.org/confluence/display/KAFKA/KIP-610%3A+Error+Reporting+in+Sink+Connectors) allows sink connectors to report individual records as being problematic, and they will be sent to the DLQ. For more information, see [Dead Letter Queue](https://docs.confluent.io/platform/6.0.0/connect/concepts.html#dead-letter-queues).

https://docs.confluent.io/platform/6.0.0/release-notes/index.html#kconnect

File: pom.xml [hunk 1]
```diff
@@ -1,142 +1,147 @@
...
-        <junit.version>5.4.2</junit.version>
-        <json.version>20141113</json.version>
-        <java.version>1.8</java.version>
-        <skipTests>false</skipTests>
-    </properties>
+  <properties>
+    <kafka.version>2.8.0</kafka.version>
+    <junit.version>5.4.2</junit.version>
+    <json.version>20141113</json.version>
+    <java.version>11</java.version>
```
Comment (danieljimeneznz) [inline pom.xml:15]: Why the bump from Java 8 to Java 11?

File: pom.xml [hunk 2]
```diff
@@ -1,142 +1,147 @@
...
 
-    <properties>
-        <kafka.version>2.2.0</kafka.version>
-        <junit.version>5.4.2</junit.version>
-        <json.version>20141113</json.version>
-        <java.version>1.8</java.version>
-        <skipTests>false</skipTests>
-    </properties>
+  <properties>
+    <kafka.version>2.8.0</kafka.version>
```
Comment (danieljimeneznz) [inline pom.xml:12]: Why the bump from Kafka 2.2.0 to 2.8.0?

File: src/main/java/nz/ac/auckland/kafka/http/sink/handler/DeadLetterQueueHandler.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package nz.ac.auckland.kafka.http.sink.handler;
+
+import nz.ac.auckland.kafka.http.sink.request.CallBackApiException;
+import org.apache.kafka.connect.sink.ErrantRecordReporter;
+
+public class DeadLetterQueueHandler implements ExceptionHandler {
+    private final ErrantRecordReporter errantRecordReporter;
+
+    public DeadLetterQueueHandler(ErrantRecordReporter errantRecordReporter) {
+
```
Comment (danieljimeneznz) [inline src/main/java/nz/ac/auckland/kafka/http/sink/handler/DeadLetterQueueHandler.java:10]:
```suggestion
```

File: src/main/java/nz/ac/auckland/kafka/http/sink/handler/DeadLetterQueueHandler.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        this.errantRecordReporter = errantRecordReporter;
+    }
+
+    @Override
+    public void handel(CallBackApiException e) {
```
Comment (danieljimeneznz) [inline src/main/java/nz/ac/auckland/kafka/http/sink/handler/DeadLetterQueueHandler.java]: `handle` spelt incorrectly?

File: src/main/java/nz/ac/auckland/kafka/http/sink/model/KafkaRecord.java [hunk 5]
```diff
@@ -4,40 +4,38 @@
...
     @Override
     public String toString() {
         return "KafkaRecord{" +
-                "topic='" + topic + '\'' +
-                ", key='" + key + '\'' +
-                ", partition=" + partition +
-                ", offset=" + offset +
-                ", value='" + value + '\'' +
+                "topic='" + record.topic() + '\'' +
+                ", key='" + record.key() + '\'' +
```
Comment (danieljimeneznz) [inline src/main/java/nz/ac/auckland/kafka/http/sink/model/KafkaRecord.java:31]: Key is no longer null safe?

File: src/main/java/nz/ac/auckland/kafka/http/sink/request/ApiRequest.java [hunk 6]
```diff
@@ -129,8 +129,10 @@ private RetryIndicator getResponseIndicator() {
         try {
             JsonObject response = new JsonParser().parse(getResponse()).getAsJsonObject();
             return response.get("retry").getAsBoolean()? RetryIndicator.RETRY : RetryIndicator.NO_RETRY ;
-        }catch (Exception ex){
-            log.warn("Json response with 'retry' field with not found. Assuming retry=true.");
+        }catch (JsonSyntaxException | IllegalStateException exception){
+            throw  exception;
+        }
+        catch (Exception ex){
```
Comment (danieljimeneznz) [inline src/main/java/nz/ac/auckland/kafka/http/sink/request/ApiRequest.java]: Change does nothing apart from remove logging

File: src/main/java/nz/ac/auckland/kafka/http/sink/request/RetryIndicator.java [hunk 7]
```diff
@@ -1,7 +1,7 @@
 package nz.ac.auckland.kafka.http.sink.request;
 
 public enum RetryIndicator {
-    RETRY(true) , NO_RETRY(false), UNKNOWN(true);
+    RETRY(true) , NO_RETRY(false), UNKNOWN(false);
```
Comment (danieljimeneznz) [inline src/main/java/nz/ac/auckland/kafka/http/sink/request/RetryIndicator.java:4]: What is changing unknown from true to false doing?

File: src/main/java/nz/ac/auckland/kafka/http/sink/handler/ResponseExceptionStrategyHandlerFactory.java [hunk 8]
```diff
@@ -9,7 +9,8 @@
         PROGRESS_BACK_OFF_DROP_MESSAGE,
         PROGRESS_BACK_OFF_STOP_TASK,
         DROP_MESSAGE,
-        STOP_TASK
+        STOP_TASK,
+        DEAD_LETTER_QUEUE
```
Comment (danieljimeneznz) [inline src/main/java/nz/ac/auckland/kafka/http/sink/handler/ResponseExceptionStrategyHandlerFactory.java:13]: Update README with new exception strategy options.

File: src/test/java/nz/ac/auckland/kafka/http/sink/request/ApiRequestTest.java [hunk 9]
```diff
@@ -116,11 +117,11 @@ void Test_sendPayload_throws_expectation_if_invalid_Json_response() throws IOExc
         String response = "{ \"message\":\"Unable to process. An unhandled exception has been thrown.\"," +
                 "   \"retry\":false";
         when(connection.getOutputStream()).thenReturn(outputStreamMock);
-        when(connection.getInputStream()).thenReturn(new ByteArrayInputStream(response.getBytes(Charset.forName("UTF-8"))));
-
-        ApiRequest apiRequest = new ApiRequest(connection,kafkaRecord);
+        when(connection.getInputStream()).thenReturn(new ByteArrayInputStream(response.getBytes(StandardCharsets.UTF_8)));
+        when(connection.getResponseCode()).thenReturn(502);
```
Comment (danieljimeneznz) [inline src/test/java/nz/ac/auckland/kafka/http/sink/request/ApiRequestTest.java:121]: Don't understand the 502 change in here?

**Review by danieljimeneznz — CHANGES_REQUESTED** [review]

**Pull request closed without merging**